subsys/nrf_security/src/drivers/nrf_oberon/Kconfig

```diff
@@ -1,353 +1,1114 @@
+												   (PSA_WANT_ALG_SHA_512 && !PSA_ACCEL_RSA8192_SHA512_PKCS1V15_SIGN)))
+
+config PSA_NEED_OBERON_RSA_PSS
+	bool
+	default y
+	depends on PSA_WANT_ALG_RSA_PSS
+	depends on PSA_NEED_OBERON_RSA_KEY_SIZE_1024 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_1536 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_2048 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_3072 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_4096 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_6144 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_8192
+
+config PSA_NEED_OBERON_RSA_PKCS1V15_SIGN
+	bool
+	default y
+	depends on PSA_WANT_ALG_RSA_PKCS1V15_SIGN
+	depends on PSA_NEED_OBERON_RSA_KEY_SIZE_1024 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_1536 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_2048 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_3072 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_4096 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_6144 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_8192
+
+config PSA_NEED_OBERON_RSA_SIGN
+	bool
+	default y
+	depends on PSA_NEED_OBERON_RSA_PSS || PSA_NEED_OBERON_RSA_PKCS1V15_SIGN
+
+config PSA_NEED_OBERON_RSA_PKCS1V15_CRYPT
+	bool
+	default y
+	depends on PSA_WANT_ALG_RSA_PKCS1V15_CRYPT
+	depends on PSA_NEED_OBERON_RSA_KEY_SIZE_1024 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_1536 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_2048 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_3072 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_4096 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_6144 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_8192
+
+config PSA_NEED_OBERON_RSA_OAEP
+	bool
+	default y
+	depends on PSA_WANT_ALG_RSA_OAEP
+	depends on PSA_NEED_OBERON_RSA_KEY_SIZE_1024 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_1536 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_2048 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_3072 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_4096 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_6144 || \
+			   PSA_NEED_OBERON_RSA_KEY_SIZE_8192
+
+config PSA_NEED_OBERON_RSA_CRYPT
+	bool
+	default y
+	depends on PSA_NEED_OBERON_RSA_PKCS1V15_CRYPT || PSA_NEED_OBERON_RSA_OAEP
+
+config PSA_NEED_OBERON_RSA_DRIVER
+	bool
+	default y
+	depends on PSA_NEED_OBERON_RSA_CRYPT || PSA_NEED_OBERON_RSA_SIGN
+
+config PSA_MAX_RSA_KEY_BITS
+	int
+	default 8192 if PSA_WANT_RSA_KEY_SIZE_8192
+	default 6144 if PSA_WANT_RSA_KEY_SIZE_6144
+	default 4096 if PSA_WANT_RSA_KEY_SIZE_4096
+	default 3072 if PSA_WANT_RSA_KEY_SIZE_3072
+	default 2048 if PSA_WANT_RSA_KEY_SIZE_2048
+	default 1536 if PSA_WANT_RSA_KEY_SIZE_1536
+	default 1024 if PSA_WANT_RSA_KEY_SIZE_1024
+	default 0
+
+# Oberon Random Driver
+
+config PSA_NEED_OBERON_CTR_DRBG_DRIVER
+	bool
+	default y
+	depends on PSA_ACCEL_ENTROPY
+	depends on PSA_USE_CTR_DRBG_DRIVER && !PSA_ACCEL_RANDOM
+
+config PSA_NEED_OBERON_HMAC_DRBG_DRIVER
+	bool
+	default y
+	depends on PSA_ACCEL_ENTROPY
+	depends on PSA_USE_HMAC_DRBG_DRIVER && !PSA_ACCEL_RANDOM
+
+endif # PSA_CRYPTO_DRIVER_OBERON
+
+# Currently Unsupported Algorithms
+
+config PSA_WANT_ALG_SHA_512_224
+	bool
+	depends on PSA_ACCEL_SHA_512_224
```

Comment:
I don't think this depends on has any effect.

It's not clear to me why it is here.

Could we remove it?

Reply:
The same applies to many options here that are promptless and default n.

Reply:
You are right, I removed all of these from Kconfig and I included them in a header file. I did that to be able to report to the user when an unsupported algorithm is enabled.